src/graphics/camera3d.rs

```diff
@@ -0,0 +1,135 @@
+use crate::glam::*;
+
+/// Camera3d bundle that holds both the `Projection` and `Camera`
+#[derive(Default, Debug, Clone, Copy)]
+pub struct Camera3dBundle {
+    /// The `Camera3d` part of this bundle
+    pub camera: Camera3d,
+    /// The `Projection` part of this bundle
+    pub projection: Projection,
+}
+
+/// A 3d Camera
+#[derive(Debug, Default, Clone, Copy)]
+pub struct Camera3d {
+    /// The position of this `Camera3d`
+    pub position: Vec3,
+    /// The yaw or y axis rotation of this `Camera3d`
+    pub yaw: f32,
+    /// The pitch or x axis rotation of this `Camera3d`
+    pub pitch: f32,
+}
+
+impl Camera3d {
+    /// Create a new camera from the given position and rotation data
+    pub fn new<V: Into<Vec3>>(position: V, yaw: f32, pitch: f32) -> Self {
+        Self {
+            position: position.into(),
+            yaw,
+            pitch,
+        }
+    }
+
+    pub(crate) fn calc_matrix(&self) -> Mat4 {
+        let (sin_pitch, cos_pitch) = self.pitch.sin_cos();
+        let (sin_yaw, cos_yaw) = self.yaw.sin_cos();
+
+        look_to_rh(
+            self.position,
+            Vec3::new(cos_pitch * cos_yaw, sin_pitch, cos_pitch * sin_yaw).normalize(),
+            Vec3::Y,
+        )
+    }
+}
+
+pub(crate) fn look_to_rh(eye: Vec3, dir: Vec3, up: Vec3) -> Mat4 {
+    let f = dir.normalize();
+    let s = f.cross(up).normalize();
+    let u = s.cross(f);
+
+    #[rustfmt::skip]
+      let mat =   Mat4::from_cols_array(&[
+            s.x, u.x, -f.x, 0.0,
+            s.y, u.y, -f.y, 0.0,
+            s.z, u.z, -f.z, 0.0,
+            -eye.dot(s), -eye.dot(u), eye.dot(f), 1.0,
+        ]
+    );
+
+    mat
+}
```

Comment:
You could probably use `glam::Mat4::look_to_rh` directly instead of having a copy of it here

Reply:
Fixed in latest commit